### PR TITLE
Return a more helpful error message for unknown db roles

### LINKED ISSUE #5939: Better Dynamic Database Allowed Role Error Message
**Describe the bug**
If a Dynamic Database Role is not allowed on a database connection, there is no useful error message or indication of the issue in the Vault server logs, audit logs, or to the end user; Just a permission denied error.

**To Reproduce**
1. Create a database mount
2. Create a database connection restricted by role
3. Create a database role does not match one of the allowed roles
4. Try to use that role.

**Expected behavior**
An error message to the requesting user or within Vault's logging that the role either isn't allowed on that database or the database configuration doesn't allow that role.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 0.10.2
* Vault CLI Version (retrieve with `vault version`): 1.0
* Server Operating System/Architecture: Docker/Ubuntu

I ran into this multiple times in our production environment that matches the versions above but I believe I also ran into it while testing locally with 1.0.

### PR DESCRIPTION
**Pull request by tyrannosaurus-becks**
Closes #5939

### CODE REVIEW

**Comment by briankassouf** [comment]
Thanks for doing this! Could you also update this test:
```
--- FAIL: TestBackend_allowedRoles (8.43s)
    backend_test.go:1160: expected error to be:permission denied got:&errors.errorString{s:"denied is not an allowed role"}
```

File: builtin/logical/database/path_creds_create.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func (b *databaseBackend) pathCredsCreateRead() framework.OperationFunc {
 		// If role name isn't in the database's allowed roles, send back a
 		// permission denied.
 		if !strutil.StrListContains(dbConfig.AllowedRoles, "*") && !strutil.StrListContainsGlob(dbConfig.AllowedRoles, name) {
-			return nil, logical.ErrPermissionDenied
+			return nil, fmt.Errorf("%s is not an allowed role", name)
```
Comment (jefferai) [inline builtin/logical/database/path_creds_create.go]: Minor preference for %q here, just to make it super clear

Comment (tyrannosaurus-becks) [reply builtin/logical/database/path_creds_create.go]: Ah yes, that would read better, thanks!

**Comment by tyrannosaurus-becks** [comment]
@briankassouf updated and pushed! The remaining test failures appear unrelated.

**Pull request merged**